rclcpp/include/rclcpp/executor.hpp

```diff
@@ -102,31 +109,26 @@ class Executor
   {
     this->add_node(node);
     // non-blocking = true
-    std::shared_ptr<AnyExecutable> any_exec;
+    AnyExecutableSharedPtr any_exec;
     while ((any_exec = get_next_executable(true))) {
       execute_any_executable(any_exec);
     }
     this->remove_node(node);
   }
 
-protected:
-  struct AnyExecutable
-  {
-    AnyExecutable()
-    : subscription(0), timer(0), callback_group(0), node(0)
-    {}
-    // Either the subscription or the timer will be set, but not both
-    rclcpp::subscription::SubscriptionBase::SharedPtr subscription;
-    rclcpp::timer::TimerBase::SharedPtr timer;
-    rclcpp::service::ServiceBase::SharedPtr service;
-    rclcpp::client::ClientBase::SharedPtr client;
-    // These are used to keep the scope on the containing items
-    rclcpp::callback_group::CallbackGroup::SharedPtr callback_group;
-    rclcpp::node::Node::SharedPtr node;
-  };
+  // Support dynamic switching of memory strategy
+  void
+  set_memory_strategy(memory_strategy::MemoryStrategySharedPtr memory_strategy)
+  {
+    if (memory_strategy == nullptr) {
```

Comment:
It's explicitly documented as a feature of `shared_ptr`, so I'd say it's ok to use if not recommended:

http://en.cppreference.com/w/cpp/memory/shared_ptr/operator_bool